src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        int numberOfFigures = 6;
+        Figure[] figures = new Figure[numberOfFigures];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < figures.length; i++) {
+            figures[i] = i > figures.length / 2
```

Comment:
You should create several random Figures, so you will use a loop. Please don't create a `new FigureSupplier()` inside the loop. Let's do it only once - before the loop starts. In this case, you are correctly creating `FigureSupplier` before the loop, but please ensure it's not created inside the loop in any other part of your code.